core/src/test/java/org/apache/iceberg/TestDataTaskParser.java

```diff
@@ -0,0 +1,147 @@
+    return "{\"task-type\":\"data-task\","
+        + "\"schema\":{\"type\":\"struct\",\"schema-id\":0,"
+        + "\"fields\":[{\"id\":1,\"name\":\"committed_at\",\"required\":true,\"type\":\"timestamptz\"},"
+        + "{\"id\":2,\"name\":\"snapshot_id\",\"required\":true,\"type\":\"long\"},"
+        + "{\"id\":3,\"name\":\"parent_id\",\"required\":false,\"type\":\"long\"},"
+        + "{\"id\":4,\"name\":\"operation\",\"required\":false,\"type\":\"string\"},"
+        + "{\"id\":5,\"name\":\"manifest_list\",\"required\":false,\"type\":\"string\"},"
+        + "{\"id\":6,\"name\":\"summary\",\"required\":false,\"type\":{\"type\":\"map\","
+        + "\"key-id\":7,\"key\":\"string\",\"value-id\":8,"
+        + "\"value\":\"string\",\"value-required\":true}}]},"
+        + "\"projection\":{\"type\":\"struct\",\"schema-id\":0,"
+        + "\"fields\":[{\"id\":1,\"name\":\"committed_at\",\"required\":true,\"type\":\"timestamptz\"},"
+        + "{\"id\":2,\"name\":\"snapshot_id\",\"required\":true,\"type\":\"long\"},"
+        + "{\"id\":3,\"name\":\"parent_id\",\"required\":false,\"type\":\"long\"},"
+        + "{\"id\":4,\"name\":\"operation\",\"required\":false,\"type\":\"string\"},"
+        + "{\"id\":5,\"name\":\"manifest_list\",\"required\":false,\"type\":\"string\"},"
+        + "{\"id\":6,\"name\":\"summary\",\"required\":false,\"type\":{\"type\":\"map\","
+        + "\"key-id\":7,\"key\":\"string\",\"value-id\":8,"
+        + "\"value\":\"string\",\"value-required\":true}}]},"
+        + "\"metadata-file\":{\"spec-id\":0,\"content\":\"DATA\","
+        + "\"file-path\":\"/tmp/metadata2.json\","
+        + "\"file-format\":\"METADATA\",\"partition\":{},"
+        + "\"file-size-in-bytes\":0,\"record-count\":2,\"sort-order-id\":0},"
+        + "\"rows\":[{\"1\":\"2009-02-13T23:31:30+00:00\",\"2\":1,\"4\":\"append\","
+        + "\"5\":\"file:/tmp/manifest1.avro\","
+        + "\"6\":{\"keys\":[\"added-data-files\",\"added-records\",\"added-files-size\",\"changed-partition-count\","
+        + "\"total-records\",\"total-files-size\",\"total-data-files\",\"total-delete-files\","
+        + "\"total-position-deletes\",\"total-equality-deletes\"],"
+        + "\"values\":[\"1\",\"1\",\"10\",\"1\",\"1\",\"10\",\"1\",\"0\",\"0\",\"0\"]}},"
+        + "{\"1\":\"2282-12-22T20:13:30+00:00\",\"2\":2,\"3\":1,\"4\":\"append\","
+        + "\"5\":\"file:/tmp/manifest2.avro\","
+        + "\"6\":{\"keys\":[\"added-data-files\",\"added-records\",\"added-files-size\",\"changed-partition-count\","
+        + "\"total-records\",\"total-files-size\",\"total-data-files\",\"total-delete-files\","
+        + "\"total-position-deletes\",\"total-equality-deletes\"],"
+        + "\"values\":[\"1\",\"1\",\"10\",\"1\",\"2\",\"20\",\"2\",\"0\",\"0\",\"0\"]}}]}";
+  }
+
+  private void assertDataTaskEquals(StaticDataTask expected, StaticDataTask actual) {
+    Assertions.assertThat(expected.schema().sameSchema(actual.schema()))
```

Comment:
```suggestion
    assertThat(expected.schema().asStruct()).isEqualTo(atual.schema().asStruct()))
```

this is slightly better, because it will actually show the difference in the schema if the assertion ever fails, whereas `.isTrue()` will only say that the schema doesn't match